### DISCUSSION TITLE
Kernel association by default

### DISCUSSION THREAD

**Discussion by crazy4pi314** [microsoft/vscode-jupyter#11075] [New Ideas]
When working in a notebook, especially when using a remote compute back-end, there are at least three places where I have to set an environment/kernel/interpret path. It is not clear to me when I launch the remote compute the state of these three settings (which remote compute/azure workspace, which Python interpreter, and which notebook kernel), or if changing one impacts the others. 

Almost always (honestly can't think of a counter example at the moment) I want all three of these to be the same/locked to each other. It _seems_ like once you have connected to a remote, the configuration of these settings is restored, but it would be nice to have an editor/extension setting to "lock" these together for sessions/editor workspace settings. Say if you set one manually, it would be great to get a notification saying: "Hey, would you like to set all these other variables the same way?"

**Comment by IanMatthewHuff** [comment]
@crazy4pi314 Kernel management is definitely something that we've had some confusion with and we're working on cleaning up, so the feedback is much appreciated here. I did have one clarifying question or two to ask. With the "lock" that you mention, was this referring more to syncing the current state of those variables in one notebook (i.e. if you change the selected kernel in the notebook kernel picker then the selected python interpreter would change as well) or more for locking one set state of those variables to then be the standard for all files in that workspace (i.e. Configure fileA.ipynb with remote compute + the kernel you want,  then "lock," then open fileB.ipynb in the same workspace and fileB.ipynb has the same configuration as fileA automatically).

I was also interested in what your expectations were regarding the link between the currently selected Python interpreter in the workspace and the kernel selected in the notebook kernel picker. We've actually been trying to move away from a link here, right now the only thing that the selected Python interpreter does for notebooks is provide a "light hint," where if you open a file where the metadata doesn't match anything on the system we suggest the currently selected python interpreter over other python interpreters as a possible notebook kernel. Was interested in what the expected link was here to see if we could handle this better.
